phpBB/includes/acp/acp_search.php

```diff
@@ -175,270 +243,235 @@ function settings($id, $mode)
+					]));
 				}
 			}
 
-			if ($updated)
-			{
-				if (method_exists($search, 'config_updated'))
-				{
-					if ($search->config_updated())
-					{
-						trigger_error($error . adm_back_link($this->u_action), E_USER_WARNING);
-					}
-				}
-			}
-
-			trigger_error($user->lang['CONFIG_UPDATED'] . $extra_message . adm_back_link($this->u_action));
+			trigger_error($this->language->lang('CONFIG_UPDATED') . $extra_message . adm_back_link($this->u_action));
 		}
 		unset($cfg_array);
 
-		$this->tpl_name = 'acp_search';
+		$this->tpl_name = 'acp_search_settings';
 		$this->page_title = 'ACP_SEARCH_SETTINGS';
 
-		$template->assign_vars([
-			'DEFAULT_SEARCH_RETURN_CHARS'	=> (int) $config['default_search_return_chars'],
-			'LIMIT_SEARCH_LOAD'				=> (float) $config['limit_search_load'],
-			'MIN_SEARCH_AUTHOR_CHARS'		=> (int) $config['min_search_author_chars'],
-			'SEARCH_INTERVAL'				=> (float) $config['search_interval'],
-			'SEARCH_GUEST_INTERVAL'			=> (float) $config['search_anonymous_interval'],
-			'SEARCH_STORE_RESULTS'			=> (int) $config['search_store_results'],
-			'MAX_NUM_SEARCH_KEYWORDS'		=> (int) $config['max_num_search_keywords'],
+		$this->template->assign_vars([
+			'DEFAULT_SEARCH_RETURN_CHARS'	=> (int) $this->config['default_search_return_chars'],
+			'LIMIT_SEARCH_LOAD'				=> (float) $this->config['limit_search_load'],
+			'MIN_SEARCH_AUTHOR_CHARS'		=> (int) $this->config['min_search_author_chars'],
+			'SEARCH_INTERVAL'				=> (float) $this->config['search_interval'],
+			'SEARCH_GUEST_INTERVAL'			=> (float) $this->config['search_anonymous_interval'],
+			'SEARCH_STORE_RESULTS'			=> (int) $this->config['search_store_results'],
+			'MAX_NUM_SEARCH_KEYWORDS'		=> (int) $this->config['max_num_search_keywords'],
 
 			'S_SEARCH_TYPES'		=> $search_options,
-			'S_YES_SEARCH'			=> (bool) $config['load_search'],
-			'S_SETTINGS'			=> true,
+			'S_YES_SEARCH'			=> (bool) $this->config['load_search'],
 
 			'U_ACTION'				=> $this->u_action . '&amp;hash=' . generate_link_hash('acp_search'),
 		]);
 	}
 
-	function index($id, $mode)
+	/**
+	 * Execute action
+	 *
+	 * @param string $id
+	 * @param string $mode
+	 * @throws Exception
+	 */
+	public function index(string $id, string $mode): void
 	{
-		global $user, $template, $phpbb_log, $request;
-		global $config, $phpbb_admin_path, $phpEx, $phpbb_container;
+		$action = $this->request->variable('action', '');
+		$state = !empty($this->config['search_indexing_state']) ? explode(',', $this->config['search_indexing_state']) : [];
 
-		$action = $request->variable('action', '');
-		$this->state = explode(',', $config['search_indexing_state']);
-
-		if ($request->is_set_post('cancel'))
-		{
-			$action = '';
-			$this->state = array();
-			$this->save_state();
-		}
-		$submit = $request->is_set_post('submit');
-
-		if (!check_link_hash($request->variable('hash', ''), 'acp_search') && in_array($action, array('create', 'delete')))
```

Comment:
No more check link hash for this?

Reply:
Yes, now is in the line 384, before it was used when is not needed, now only in the indexing progress